### PR TITLE
Use go 1.13.8 in Dockerfile

### PR DESCRIPTION
**Pull request by uthark**
Use go 1.13 in Dockerfile to build the image.

Performed testing:
1. Built new docker image with new version.
2. Performed rollout upgrade using new version of upgrade manager.
3. No errors / crashes with new image.

### CODE REVIEW

**Comment by uthark** [comment]
@shrinandj @eytan-avisror could you please take a look?

**Comment by eytan-avisror** [comment]
Hi @uthark 
Can you add a `testing done` section to your PR?

**Comment by uthark** [comment]
@eytan-avisror Done.

**Comment by eytan-avisror** [comment]
> @eytan-avisror Done.

I mean a section that details the testing you have done to verify this won't have any negative impact.

Such as a basic functional test

**Comment by eytan-avisror** [comment]
For example, build this controller with go 1.13.8, run it, do a rolling upgrade, etc.

**Comment by uthark** [comment]
@eytan-avisror Clarified testing - I deployed new version and performed rolling update using it.

**Comment by eytan-avisror** [comment]
Can you update semaphore as well?
I think we are using an older version there:
https://github.com/keikoproj/upgrade-manager/blob/1878e36bbfb93d3c9846dee02da5a00d7690b62d/.semaphore/semaphore.yml#L11

File: .semaphore/setup.sh [hunk 1]
```diff
@@ -1,8 +1,8 @@
 cache restore $SEMAPHORE_PROJECT_NAME-dep-$(checksum .semaphore/setup.sh)
 # checks if the packages are already installed
-if [ ! -d '/packages' -o ! -f '/packages/go1.13.6.linux-amd64.tar.gz' ]; then
+if [ ! -d '/packages' -o ! -f '/packages/go1.13.8.linux-amd64.tar.gz' ]; then
```
Comment (uthark) [inline .semaphore/setup.sh]: One thing I don't understand is why there is a go installation in setup. I thought semaphore/golang should provide it already. Do you know why it was done this way?

Comment (eytan-avisror) [reply .semaphore/setup.sh]: Yeah, I think just using the appropriate golang image should be sufficient.
@kianjones4 any idea why we need to run `setup.sh` at all?

Comment (uthark) [reply .semaphore/setup.sh]: @eytan-avisror setup.sh also caches kube-builder.

Comment (uthark) [reply .semaphore/setup.sh]: Cleaning this up can be a separate PR though.

Comment (eytan-avisror) [reply .semaphore/setup.sh]: we don't really need kubebuilder in this context - only controller-gen.
Perhaps setup.sh can be removed in future PRs all together

**Pull request merged**